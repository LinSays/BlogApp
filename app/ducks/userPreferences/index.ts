import { actions, namespace, reducer } from "./userPreferences.slice";
import selectors from "./userPreferences.selectors";

export default { actions, namespace, reducer, selectors };
