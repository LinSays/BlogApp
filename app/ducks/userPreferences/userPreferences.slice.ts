import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { ElectricityType } from "carbon-footprint";

import { currentLanguage } from "utils";

const initialState = {
  acceptedTermsOfUseVersion: 0,
  activatedNotifications: false,
  location: ElectricityType.world,
  language: currentLanguage,
};

const userPreferences = createSlice({
  name: "userPreferences",
  initialState,
  reducers: {
    acceptTermsOfUse(state, action: PayloadAction<number>) {
      state.acceptedTermsOfUseVersion = action.payload;
    },
    toggleNotifications(state, action: PayloadAction<boolean>) {
      state.activatedNotifications = action.payload;
    },
    updateLocation(state, action: PayloadAction<ElectricityType>) {
      state.location = action.payload;
    },
    changeLanguage(state, action: PayloadAction<string>) {
      state.language = action.payload;
    },
  },
});

const {
  acceptTermsOfUse,
  updateLocation,
  toggleNotifications,
  changeLanguage,
} = userPreferences.actions;

export const actions = {
  acceptTermsOfUse,
  updateLocation,
  toggleNotifications,
  changeLanguage,
};

export const namespace = userPreferences.name;

export const reducer = userPreferences.reducer;
