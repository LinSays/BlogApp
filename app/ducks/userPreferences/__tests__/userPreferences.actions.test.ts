import { ElectricityType } from "carbon-footprint";

import userPreferences from "../";

describe("userPreferences actions should", () => {
  it("export expected actions", () => expect(userPreferences.actions).toMatchSnapshot());

  it("be able to update to a new term of use version", () => {
    const acceptedTermsOfUseVersion = 2;

    const expectedAction = {
      type: userPreferences.actions.acceptTermsOfUse.toString(),
      payload: acceptedTermsOfUseVersion,
    };
    expect(userPreferences.actions.acceptTermsOfUse(acceptedTermsOfUseVersion)).toEqual(
      expectedAction
    );
  });

  it("be able to update to update user location", () => {
    const expectedAction = {
      type: userPreferences.actions.updateLocation.toString(),
      payload: ElectricityType.france,
    };
    expect(userPreferences.actions.updateLocation(ElectricityType.france)).toEqual(expectedAction);
  });

  it("be able to update to update notifications", () => {
    const expectedAction = {
      type: userPreferences.actions.toggleNotifications.toString(),
      payload: true,
    };
    expect(userPreferences.actions.toggleNotifications(true)).toEqual(expectedAction);
  });
});
