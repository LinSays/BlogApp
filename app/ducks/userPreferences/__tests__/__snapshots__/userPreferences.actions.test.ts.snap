// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userPreferences actions should export expected actions 1`] = `
Object {
  "acceptTermsOfUse": [Function],
  "changeLanguage": [Function],
  "toggleNotifications": [Function],
  "updateLocation": [Function],
}
`;
